core/src/main/java/com/datastrato/gravitino/metrics/source/MetricsSource.java

```diff
@@ -85,6 +94,21 @@ public Histogram getHistogram(String name) {
    * @return a new or pre-existing Timer
    */
   public Timer getTimer(String name) {
-    return this.metricRegistry.timer(name);
+    return this.metricRegistry.timer(
+        name,
+        () ->
+            new Timer(
+                new SlidingTimeWindowArrayReservoir(
+                    getTimeSlidingWindowSeconds(), TimeUnit.SECONDS)));
+  }
+
+  protected int getTimeSlidingWindowSeconds() {
+    MetricsSystem metricsSystem = GravitinoEnv.getInstance().metricsSystem();
```

Comment:
It's a little hacky here, couldn't find a more general way to inject configurations.

Reply:
Why can't you pass the config value from metrics system to metrics source?

Reply:
currently, creating metrics source is depend of MetricsSystem, so couldn't pass config value to metrics source.
One possible solution is creating MetricsSource using MetricsSystem or Config.

Reply:
passing `Config` to `MetricsSource`, please review again